### PR TITLE
optional country code

### PR DESCRIPTION
**Pull request by tristanhcole**
Installed [react-phone-input-2](https://www.npmjs.com/package/react-phone-input-2) as it's a react package, lightweight (406 kB), easy to style and implement using our current redux-form setup.

Utilizes active organisation country code so that you don't need to add the country code when creating a new user.

NOTE: This doesn't include search as will be implemented server-side on @michielderoos PR

### CODE REVIEW

**Comment by tristanhcole** [comment]
[optional country codes](https://trello.com/c/XvUgYs4c/268-optional-country-codes)

File: app/client/components/form/FormValidation.ts [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
+      let countryCode;
+      let orgId;
+      try {
+        orgId = store.getState().login.organisationId;
+        countryCode =
+          orgId && store.getState().organisations.byId[orgId].country_code;
+      } catch (e) {
+        console.log("Something went wrong", e);
+        countryCode = undefined;
+        Sentry.captureException(e);
```
Comment (tristanhcole) [inline app/client/components/form/FormValidation.ts:19]: adding sentry to capture potential errors on this. 

Using `undefined` so validation falls back to enforcing a country code

**Comment by tristanhcole** [comment]
I was thinking last night it could make sense to have the country code in small text underneath the PhoneInput

**Comment by tristanhcole** [comment]
> I'm a little confused as to what this PR is supposed to do... it doesn't actually send the phone number with the country code to the backend?

It's just frontend form validation so that you can enter a phone number without the country code (the backend will then just use the current org country code)

**Comment by WillRuddick** [comment]
just to note that without the country code people put a 0 in front to call
someone. Would be good to watch out for that.

**Comment by enjeyw** [comment]
> I was thinking last night it could make sense to have the country code in small text underneath the PhoneInput

I think we need to do _something_ (like maybe have the extension at the front with a dropdown) - at the moment if i enter 0412-888-888 I have no idea what that's gonna get parsed as.

**Comment by enjeyw** [comment]
> > I was thinking last night it could make sense to have the country code in small text underneath the PhoneInput
> 
> I think we need to do _something_ (like maybe have the extension at the front with a dropdown) - at the moment if i enter 0412-888-888 I have no idea what that's gonna get parsed as.

Any update on this @tristanhcole ?

**Comment by tristanhcole** [comment]
> > > I was thinking last night it could make sense to have the country code in small text underneath the PhoneInput
> > 
> > 
> > I think we need to do _something_ (like maybe have the extension at the front with a dropdown) - at the moment if i enter 0412-888-888 I have no idea what that's gonna get parsed as.
> 
> Any update on this @tristanhcole ?

No, I haven't done anything else on this yet. Unsure what would be best... a dropdown like you described?

**Comment by enjeyw** [comment]
> > > > I was thinking last night it could make sense to have the country code in small text underneath the PhoneInput
> > > 
> > > 
> > > I think we need to do _something_ (like maybe have the extension at the front with a dropdown) - at the moment if i enter 0412-888-888 I have no idea what that's gonna get parsed as.
> > 
> > 
> > Any update on this @tristanhcole ?
> 
> No, I haven't done anything else on this yet. Unsure what would be best... a dropdown like you described?

I'm not sure either, but i think dropdown is the standard practice for international numbers?

**Comment by michielderoos** [comment]
[This](https://github.com/jackocnr/intl-tel-input) telephone input library might do the trick

**Comment by tristanhcole** [comment]
> [This](https://github.com/jackocnr/intl-tel-input) telephone input library might do the trick

Looks great, will look at implementing :)

**Comment by tristanhcole** [comment]
@michielderoos Installed [react-phone-input-2](https://www.npmjs.com/package/react-phone-input-2) as it's a react package, lightweight (406 kB) and easy to style and implement using our current redux-form setup.

**Comment by tristanhcole** [comment]
> Looks great! QA'd, and seems to work well for a good batch of phone numbers I've tried. One tiny nit is that it doesn't re-evaluate a phone number validity on character-entry, so after failing, a valid phone number doesn't "pass" until you change form fields. I think that's fine though, more a personal preference.
> image

Thanks! Yeah, that's how redux-form is setup currently.

**Pull request merged**